cmd/istanbul/setup/cmd.go

```diff
@@ -84,19 +89,29 @@ func gen(ctx *cli.Context) error {
 			str, _ := json.MarshalIndent(v, "", "\t")
 			fmt.Println(string(str))
 		}
+
+		if ctx.Bool(saveFlag.Name) {
+			folderName := strconv.Itoa(i)
+			os.MkdirAll(folderName, os.ModePerm)
+			ioutil.WriteFile(path.Join(folderName, "nodekey"), []byte(nodekeys[i]), os.ModePerm)
+		}
 	}
 
 	if ctx.Bool(verboseFlag.Name) {
 		fmt.Print("\n\n\n")
 	}
 
+	staticNodes, _ := json.MarshalIndent(nodes, "", "\t")
 	if ctx.Bool(staticNodesFlag.Name) {
-		staticNodes, _ := json.MarshalIndent(nodes, "", "\t")
 		fmt.Println("static-nodes.json")
 		fmt.Println(string(staticNodes))
 		fmt.Print("\n\n\n")
 	}
 
+	if ctx.Bool(saveFlag.Name) {
```

Comment:
This should be moved into 
```
if ctx.Bool(staticNodesFlag.Name) {
        ...
        if ctx.Bool(saveFlag.Name) {
                ...
        }
}
```

If users didn't use `--nodes` flag, then `static-nodes.json` is not necessary to be generated. It's my thought.
`static-nodes.json` is not mandatory but `genesis.json` is.

Reply:
OK